From 5049606e637d2867313fd64bffd27dcdc049135e Mon Sep 17 00:00:00 2001
From: Vincent ABRIOU <vincent.abriou@st.com>
Date: Tue, 26 Apr 2022 09:52:26 +0200
Subject: [PATCH 3/6] TFLite: cmake: force tensorflow-lite shared library

Generate libtensorflow-lite.so library by keeping static library
dependencies. The aim of this step is to simplify the link of applications
to a single dynamic library.

Signed-off-by: Othmane AHL ZOUAOUI <othmane.ahlzouaoui@st.com>
Signed-off-by: Vincent ABRIOU <vincent.abriou@st.com>
---
 tensorflow/lite/CMakeLists.txt                              | 2 +-
 .../lite/tools/pip_package/build_pip_package_with_cmake.sh  | 6 +++++-
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/tensorflow/lite/CMakeLists.txt b/tensorflow/lite/CMakeLists.txt
index 200d1a7c46d..e2253491713 100644
--- a/tensorflow/lite/CMakeLists.txt
+++ b/tensorflow/lite/CMakeLists.txt
@@ -458,7 +458,7 @@ include_directories(
 )
 
 # TFLite library
-add_library(tensorflow-lite
+add_library(tensorflow-lite SHARED EXCLUDE_FROM_ALL
   ${TFLITE_CORE_API_SRCS}
   ${TFLITE_CORE_SRCS}
   ${TFLITE_C_SRCS}
diff --git a/tensorflow/lite/tools/pip_package/build_pip_package_with_cmake.sh b/tensorflow/lite/tools/pip_package/build_pip_package_with_cmake.sh
index 069bb8d0f4c..c2bfd07d713 100755
--- a/tensorflow/lite/tools/pip_package/build_pip_package_with_cmake.sh
+++ b/tensorflow/lite/tools/pip_package/build_pip_package_with_cmake.sh
@@ -132,7 +132,7 @@ case "${TENSORFLOW_TARGET}" in
     ;;
 esac
 
-cmake --build . --verbose -j 2 -t _pywrap_tensorflow_interpreter_wrapper benchmark_model
+cmake --build . --verbose -j 2 -t tensorflow-lite _pywrap_tensorflow_interpreter_wrapper benchmark_model tensorflow-lite
 cd "${BUILD_DIR}"
 
 case "${TENSORFLOW_TARGET}" in
@@ -146,10 +146,14 @@ esac
 
 cp "${BUILD_DIR}/cmake_build/_pywrap_tensorflow_interpreter_wrapper${LIBRARY_EXTENSION}" \
    "${BUILD_DIR}/tflite_runtime"
+cp "${BUILD_DIR}/cmake_build/libtensorflow-lite${LIBRARY_EXTENSION}" \
+   "${BUILD_DIR}"
+
 # Bazel generates the wrapper library with r-x permissions for user.
 # At least on Windows, we need write permissions to delete the file.
 # Without this, setuptools fails to clean the build directory.
 chmod u+w "${BUILD_DIR}/tflite_runtime/_pywrap_tensorflow_interpreter_wrapper${LIBRARY_EXTENSION}"
+chmod u+w "${BUILD_DIR}/libtensorflow-lite${LIBRARY_EXTENSION}"
 
 # Build python wheel.
 cd "${BUILD_DIR}"
-- 
2.25.1

